proxy/InkAPI.cc

```diff
@@ -4858,6 +4858,15 @@ TSHttpTxnInfoIntGet(TSHttpTxn txnp, TSHttpTxnInfoKey key, TSMgmtInt *value)
   return TS_SUCCESS;
 }
 
+bool
+TSHttpTxnIsWebsocket(TSHttpTxn txnp)
+{
+  sdk_assert(sdk_sanity_check_txn(txnp) == TS_SUCCESS);
+
+  HttpSM *sm = (HttpSM *)txnp;
+  return sm->t_state.is_upgrade_request && sm->t_state.upgrade_token_wks == MIME_VALUE_WEBSOCKET;
```

Comment:
You should probably instead do, as is_websocket was set if all the required websocket headers were sent, also there is a check for the upgrade succeeding.

return sm->t_state.is_upgrade_request && sm->t_state.did_upgrade_succeed && sm->t_state.is_websocket